test/ref_generators/generate_dyn_slice_ref.py

```diff
@@ -0,0 +1,626 @@
+                   const AxisSet& ub_mask,
+                   const AxisSet& new_mask,
+                   const AxisSet& shrink_mask,
+                   const AxisSet& ellipsis_mask,
+                   const Shape& expected_output_shape,
+                   const std::vector<T>& expected_values)
+{
+    auto arg = std::make_shared<op::Parameter>(input_element_type, input_shape);
+    auto lb = std::make_shared<op::Parameter>(element::i64, Shape{lb_values.size()});
+    auto ub = std::make_shared<op::Parameter>(element::i64, Shape{ub_values.size()});
+    auto strides = std::make_shared<op::Parameter>(element::i64, Shape{strides_values.size()});
+
+    std::vector<T> input_values(shape_size(input_shape));
+    std::iota(input_values.begin(), input_values.end(), static_cast<T>(0));
+
+    auto slice = std::make_shared<op::DynSlice>(arg, lb, ub, strides, lb_mask, ub_mask, new_mask, shrink_mask, ellipsis_mask);
+
+    auto f = std::make_shared<Function>(NodeVector{slice}, ParameterVector{arg, lb, ub, strides});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}",true);
+    auto ex = backend->compile(f);
+
+    auto input_arg = backend->create_tensor(input_element_type, input_shape);
+    auto input_lb = backend->create_tensor(element::i64, Shape{lb_values.size()});
+    auto input_ub = backend->create_tensor(element::i64, Shape{ub_values.size()});
+    auto input_strides = backend->create_tensor(element::i64, Shape{strides_values.size()});
+    copy_data(input_arg, input_values);
+    copy_data(input_lb, lb_values);
+    copy_data(input_ub, ub_values);
+    copy_data(input_strides, strides_values);
+
+    auto output = backend->create_dynamic_tensor(input_element_type, PartialShape::dynamic());
+
+    ex->call_with_validate({output}, {input_arg, input_lb, input_ub, input_strides});
+
+    EXPECT_EQ(output->get_element_type(), input_element_type);
+    EXPECT_EQ(output->get_shape(), expected_output_shape);
+
+    auto output_values = read_vector<T>(output);
+
+    EXPECT_EQ(output_values, expected_values);
+}
+''')
+
+def write_footer(f):
+    f.write('''\
+// clang-format on
+''')
+
+
+def main():
+    if len(sys.argv) < 2:
+        sys.stderr.write('Output filename is required\n')
+        sys.exit(1)
+
+    f = open(sys.argv[1], 'w')
+    write_header(f)
+
+    t = SliceTestWriter(stream=f)
+
+    t.set_shape((4,))
+    for dt in ['int32','int64','float32','uint32']:
+        t.set_dtype(dt)
+
+        t[np.newaxis,3:0:-1]
+        t[...]
+        t[1:3]
+        t[2]
+        t[3:0:-2]
+        t[3::-2]
+        t[4::-2]
+        t[5::-2]
+        t[-9000:-8000:2]
+        t[-9000:8000:2]
+        t[-5:5:2]
+        t[np.newaxis]
+        t[np.newaxis,np.newaxis]
+        t[np.newaxis,np.newaxis,...,np.newaxis]
+
+        t.set_shape((5,))
+        t[3:0:-2]
+        t[0:3:2]
+        t[0:4:2]
+        t[0:5:2]
+        t[0:6:2]
+        t[0:100:2]
+        t[4:0:-2]
+        t[4:0:-3]
+        t[3:2:1]
+        t[4::-2]
```

Comment:
as far as indexing in numpy goes here are 2 possibilities:
1. stop is negative, stride is negative
`t[1:-5:-1] ---> (2,)`
`t[1:-1:-1] ---> (0,)` . (empty)

2. None
`t[1:None] ---> (3,)` . (None means end)
`x[1:None:-1] ---> (2,)`

Reply:
Thanks @sayantan-nervana. Added these, and they all pass :)